### PR TITLE
Move `machine-controller-manager-provider-local` into `gardener/gardener`

### PR DESCRIPTION
**Pull request by timebertt**
**How to categorize this PR?**

/area dev-productivity
/kind enhancement

**What this PR does / why we need it**:
In order to improve development productivity, this PR moves the content of the https://github.com/gardener/machine-controller-manager-provider-local repository into `gardener/gardener`. The images are built dynamically with Skaffold and pushed into the local registry.

On the way, we
- renamed the images in the `skaffold.yaml` file (removing the `europe-docker.pkg.dev/*` prefixes since these images are never actually pushed to this real remote registry
- fixed a bug in the `logcheck` `golangci-lint` plugin to prevent false positive findings in case a package-level functions disguises as `logr` method
- removed the entrypoint of the provider-specific machine-controller sidecars in the helper functions used by extensions to remove assumptions of how the image has been built

**Release note**:

```other operator
NONE
```

### CODE REVIEW

File: docs/extensions/machine-controller-provider-local.md [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+Out of tree (controller-based) implementation for `local` as a new provider.
+The local out-of-tree provider implements the interface defined at [MCM OOT driver](https://github.com/gardener/machine-controller-manager/blob/master/pkg/util/provider/driver/driver.go).
+
+## Fundamental Design Principles
+
+Following are the basic principles kept in mind while developing the external plugin.
+
+- Communication between this Machine Controller (MC) and Machine Controller Manager (MCM) is achieved using the Kubernetes native declarative approach.
+- Machine Controller (MC) behaves as the controller used to interact with the cloud provider AWS and manage the VMs corresponding to the machine objects.
```
Comment (plkokanov) [inline docs/extensions/machine-controller-provider-local.md]:
```suggestion
- Machine Controller (MC) behaves as the controller used to interact with the `local` provider and manage the VMs corresponding to the machine objects.
```
Or something along those lines.

File: skaffold.yaml [hunk 2]
```diff
@@ -1075,7 +1124,7 @@ build:
         ldflags:
           - '{{.LD_FLAGS}}'
         main: ./cmd/gardenlet
-    - image: europe-docker.pkg.dev/gardener-project/releases/gardener/resource-manager
+    - image: gardener-resource-manager
```
Comment (plkokanov) [inline skaffold.yaml]: I think these changes won't work with the [provider extensions dev setup](https://github.com/gardener/gardener/blob/master/docs/deployment/getting_started_locally_with_extensions.md).
There are kyverno policies which expect the image name to match: https://github.com/gardener/gardener/blob/fa31ef3ee09b5d5fd15666617348c0ac3471f182/example/provider-extensions/kyverno-policies/gardener-image-pull-policy.yaml#L25

If you also update the policy, note that we plan to also use them for the remote extension setup for some extension controllers, e.g:  https://github.com/gardener/gardener-extension-registry-cache/pull/193

**Review by plkokanov — CHANGES_REQUESTED** [review]

File: skaffold.yaml (hunk 2 above)
Comment (timebertt) [reply skaffold.yaml]: Good hint, thanks. I dropped the image name simplification from this PR and went back to fully qualified image refs for the new images.

**Comment by timebertt** [comment]
Thanks for your review @plkokanov. I rebased and addressed your suggestions.
We will take a look at the failing tests later today.

File: skaffold.yaml (hunk 2 above)
Comment (timebertt) [reply skaffold.yaml]: @rfranzke went back to the simplified image names but with a well-defined prefix `local-skaffold/`. https://github.com/gardener/gardener-extension-registry-cache/pull/193 should be adapted accordingly.

**Pull request merged**